runtime/src/current_sandbox.c

```diff
@@ -212,6 +212,9 @@ current_sandbox_fini()
 	assert(sandbox->state == SANDBOX_RUNNING_SYS);
 	sandbox_close_http(sandbox);
 	sandbox_set_as_returned(sandbox, SANDBOX_RUNNING_SYS);
+	if (module_is_paid(sandbox->module)) {
```

Comment:
You have replicated this code around 5 times. Seems like in the long term, it should be a function.